recipes-kernel/linux/files/bcm2712-raspberrypi5-usb.dts

```diff
@@ -0,0 +1,262 @@
+/dts-v1/;
+
+/ {
+	#address-cells = <0x2>;
+	#size-cells = <0x2>;
+
+	gic: gic {
+		#interrupt-cells = <0x3>;
+		interrupt-controller;
+	};
+
+	passthrough {
+		compatible = "raspberrypi,5-model-b", "brcm,bcm2712", "simple-bus";
+		ranges;
+		#address-cells = <0x2>;
+		#size-cells = <0x2>;
+
+		clk_xosc: clk_xosc {
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			clock-output-names = "xosc";
+			clock-frequency = <50000000>;
+		};
+
+		macb_pclk: macb_pclk {
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			clock-output-names = "pclk";
+			clock-frequency = <200000000>;
+		};
+
+		macb_hclk: macb_hclk {
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			clock-output-names = "hclk";
+			clock-frequency = <200000000>;
+		};
+
+		mip0: msi-controller@1000130000 {
+			compatible = "brcm,bcm2712-mip-intc";
+			reg = <0x10 0x00130000 0x0 0xc0>;
+			interrupt-parent = <&gic>;
+			msi-controller;
+			interrupt-controller;
+			#interrupt-cells = <2>;
+			brcm,msi-base-spi = <128>;
+			brcm,msi-num-spis = <64>;
+			brcm,msi-offset = <0>;
+			brcm,msi-pci-addr = <0xff 0xfffff000>;
+			xen,path = "/axi/msi-controller@130000";
+			xen,force-assign-without-iommu = <0x1>;
+			xen,reg = <0x10 0x00130000 0x0 0x1000 0x10 0x00130000>;
+			interrupts = <0x00 0x80 0x04 0x00 0x81 0x04 0x00 0x82 0x04
+				      0x00 0x83 0x04 0x00 0x84 0x04 0x00 0x85 0x04
+				      0x00 0x86 0x04 0x00 0x87 0x04 0x00 0x88 0x04
+				      0x00 0x89 0x04 0x00 0x8a 0x04 0x00 0x8b 0x04
+				      0x00 0x8c 0x04 0x00 0x8d 0x04 0x00 0x8e 0x04
+				      0x00 0x8f 0x04 0x00 0x90 0x04 0x00 0x91 0x04
+				      0x00 0x92 0x04 0x00 0x93 0x04 0x00 0x94 0x04
+				      0x00 0x95 0x04 0x00 0x96 0x04 0x00 0x97 0x04
+				      0x00 0x98 0x04 0x00 0x99 0x04 0x00 0x9a 0x04
+				      0x00 0x9b 0x04 0x00 0x9c 0x04 0x00 0x9d 0x04
+				      0x00 0x9e 0x04 0x00 0x9f 0x04 0x00 0xa0 0x04>;
+		};
+
+		pciex: pcie@1000120000 {
+			compatible = "brcm,bcm2712-pcie";
+			#address-cells = <0x3>;
+			#size-cells = <0x2>;
+			#interrupt-cells = <0x1>;
+			reg = <0x10 0x120000 0x0 0x9310>;
+			aspm-no-l0s;
+			brcm,vdm-qos-map = <0xbbaa9888>;
+			brcm,enable-mps-rcb;
+			brcm,enable-l1ss;
+			dma-ranges = <0x2000000 0x0 0x0 0x1f 0x0 0x0 0x400000
+				      0x43000000 0x10 0x0 0x0 0x0 0x10 0x0>;
+			ranges = <0x2000000 0x0 0x0 0x1f 0x0 0x0 0xfffffffc
+				  0x43000000 0x4 0x0 0x1c 0x0 0x3 0x0>;
+			interrupt-parent = <&gic>;
+			/*
+			 * HACK: But the mip0: msi-controller@130000 doesn't have IRQs defined
```

Comment:
I don't think we need this hack in the partial device tree

Reply:
no. I'd like it to stay here as there is no other place for it - and It clearly identifies WHY our DT is differ from original RPI5 DT.

Reply:
I can understand why do you need this in the Xen device tree. But why do you need this in DomD DTS?

Reply:
It's hack for DomD DT, as in dom0less mode it makes PCIe IRQ work properly.